### PR TITLE
Use Arel to constuct INSERT, UPDATE and DELETE statements

### PR DESCRIPTION
**Pull request by betesh-at-data-axle**
This will make it easier to add another column to the INSERT statement

### CODE REVIEW

File: lib/superstore/adapters/jsonb_adapter.rb [hunk 1]
```diff
@@ -43,48 +48,31 @@ def update(table, id, attributes)
...
+        statement.where(Arel::Nodes::SqlLiteral.new(primary_key_column).eq(id))
+
         value_update = superstore_column
         nil_properties.each do |property|
           value_update = "(#{value_update} - '#{property}')"
         end
 
         if not_nil_attributes.any?
-          value_update = "(#{value_update} || #{to_quoted_jsonb(not_nil_attributes)})"
+          value_update = "(#{value_update} || #{quote(to_quoted_jsonb(not_nil_attributes))})"
```
Comment (matthinea) [inline lib/superstore/adapters/jsonb_adapter.rb:61]: It's a little confusing that `quote` is necessary here when the method name is already `to_quoted_jsonb`.

File: lib/superstore/adapters/jsonb_adapter.rb [hunk 2]
```diff
@@ -43,48 +48,31 @@ def update(table, id, attributes)
...
         end
 
-        statement = "UPDATE #{table} SET #{superstore_column} = #{value_update} WHERE #{primary_key_column} = #{quote(id)}"
-
-        execute statement
-      end
-
-      def delete(table, ids)
```
Comment (matthinea) [inline lib/superstore/adapters/jsonb_adapter.rb:60]: Where did the delete method go?

Comment (betesh-at-data-axle) [reply lib/superstore/adapters/jsonb_adapter.rb:60]: `Superstore::Persistence` doesn't override [_delete_record](https://github.com/rails/rails/blob/v6.1.5/activerecord/lib/active_record/persistence.rb#L388).

File: lib/superstore/adapters/jsonb_adapter.rb (hunk 1 above)
Comment (betesh-at-data-axle) [reply lib/superstore/adapters/jsonb_adapter.rb]: Renamed

**Review by matthinea — APPROVED** [review]

**Pull request merged**